File: backend/src/model/Domain/authorization.go
package Domain

import (
	"database/sql"
	"time"
)

type Authorization struct {
	ID             uint32       `gorm:"primaryKey" json:"id,omitempty"`
	UserID         string       `gorm:"primaryKey" json:"user_id,omitempty"`
	AuthKB         string       `json:"auth_kb,omitempty"`
	RequestStateKB string       `json:"request_state_kb,omitempty"`
	RequestDate    sql.NullTime `json:"request_date,omitempty"`
	AppliedDate    sql.NullTime `json:"applied_date,omitempty"`
	CreatedAt      time.Time    `gorm:"autoCreateTime" json:"created_at,omitempty"`
	UpdatedAt      time.Time    `gorm:"autoUpdateTime;datetime(0)" json:"updated_at,omitempty"`
}

File: totonoe-app/src/@types/Authorization.ts
export type AuthState = {
    /**
     * 申請区分(999: 管理者, 1: 投稿可能, 0: 一般ユーザー)
     */
    auth_kb: string,

    /**
     * 申請状態(0:未申請, 1:申請中, 2: 申請完了, 3: 申請破棄)
     */
    request_state_kb: string,
}